File: lcm-gcd/lcm-gcd.go
package main

import (
	"fmt"
)

/**
最小公倍数
 */
func lcm(a int, b int) int {
	return a * b / gcd(a, b)
}

/**
最大公約数
 */
func gcd(a int, b int) int {
	if b == 0 {
		return a
	}
	return gcd(b, a % b)
}

func main() {
	fmt.Println(lcm(6, 4))
}

File: lcm-gcd/lcm-gcd.php
<?php

/**
 * 最大公約数
 * @param $a
 * @param $b
 * @return int
 */
function gcd($a, $b)
{
    if ($b === 0) return (int)$a;
    return gcd($b, $a % $b);
}

/**
 * 最小公倍数
 * @param $a
 * @param $b
 * @return int
 */
function lcm($a, $b)
{
    return (int)($a * $b / gcd($a, $b));
}